### DISCUSSION TITLE
I wonder why the invalid property in the fieldState is deprecated.

### DISCUSSION THREAD

**Discussion by mango906** [react-hook-form/react-hook-form#9750] [Q&A]
I wonder why the invalid property in the fieldState is deprecated.

I check this discussions.
https://github.com/react-hook-form/react-hook-form/discussions/8983

But, I think invalid and error seem to behave differently.

For example, if i do not enter a required field in the first rendering, this field is not valid, but there is also no error. The error value changes only when there is an action. Therefore, I think the invalid property should be maintained.

I would like to ask you again if the error property of fieldState can replace the invalid property.

**Comment by bluebill1049** [comment]
I think you are actually right. Would you like to send a PR to fix this (revert the deprecate)?

**Reply by mango906** [reply]
Okay. I will work on that part.

When I check code, the behavior of invalid and error is actually the same.

```typescript
 const getFieldState: UseFormGetFieldState<TFieldValues> = (
    name,
    formState,
  ) => ({
    invalid: !!get((formState || _formState).errors, name),
    isDirty: !!get((formState || _formState).dirtyFields, name),
    isTouched: !!get((formState || _formState).touchedFields, name),
    error: get((formState || _formState).errors, name),
  });
```

As mentioned earlier, invalid should be classified by whether the field is valid or not, not by error. 
It would be nice to fix this as well.

**Reply by mango906** [reply]
Could you help me with this?

**Reply by mango906** [reply]
I make PR. But I need a help from you. @bluebill1049 

https://github.com/react-hook-form/react-hook-form/pull/9755

**Reply by bluebill1049** [reply]
> As mentioned earlier, invalid should be classified by whether the field is valid or not, not by error.
It would be nice to fix this as well.

You are right, but in order for us to support such a prop we will have to go back to the drawing board to start supporting a new formState which is `inValidFields` and this has to be a global state to start with, so perhaps this will need a feature request to start with and be planed proper and we can leave the deprecate annotation for now then. I will cover your issue into a feature request and pin into the project board.

**Reply by bluebill1049** [reply]
https://github.com/react-hook-form/react-hook-form/issues/9757

**Reply by mango906** [reply]
Okay. I would appreciate it if you could let me know when you make any progress. Thanks

**Reply by bluebill1049** [reply]
Will do, you can subscribe to that feature request, i will leave update there.
